cfme/tests/cloud_infra_common/test_html5_vm_console.py

```diff
@@ -0,0 +1,168 @@
+"""Test for HTML5 Remote Consoles of VMware/RHEV/RHOSP Providers."""
+import pytest
+import imghdr
+import re
+import time
+
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.configure.configuration import VMwareConsoleSupport
+from cfme.infrastructure.provider.rhevm import RHEVMProvider
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.configure import configuration
+from cfme.common.vm import VM
+from utils import testgen, version, ssh
+from utils.appliance.implementations.ui import navigate_to
+from utils.log import logger
+from utils.conf import credentials
+from wait_for import wait_for
+
+pytestmark = pytest.mark.usefixtures('setup_provider')
+
+pytest_generate_tests = testgen.generate(
+    [OpenStackProvider, RHEVMProvider, VMwareProvider],
+    scope='module'
+)
+
+
+@pytest.fixture(scope="function")
+def vm_obj(request, provider, setup_provider, small_template, vm_name):
+    """
+    Create a VM on the provider with the given template, and return the vm_obj.
+
+    Also, it will remove VM from provider using nested function _delete_vm
+    after the test is completed.
+
+    """
+    vm_obj = VM.factory(vm_name, provider, template_name=small_template)
+
+    @request.addfinalizer
+    def _delete_vm():
+        try:
+            vm_obj.delete_from_provider()
+        except Exception:
+            logger.warning("Failed to delete vm `{}`.".format(vm_obj.name))
+
+    vm_obj.create_on_provider(timeout=2400, find_in_cfme=True, allow_skip="default")
+    return vm_obj
+
+
+def _configure_vmware_console_for_test(appliance):
```

Comment:
**REQUIRED:** I think we should have a method on the Appliance, in the configuration page called something like ```set_console_mode``` and probably with its counterpart, ```get_console_mode```, It then allows other people to use the function you have written here, currently it's hidden.